src/rust/engine/src/nodes.rs

```diff
@@ -212,17 +220,17 @@ impl Select {
     variant_value: Option<&str>
   ) -> Option<Value> {
     // Check whether the subject is-a instance of the product.
-    if let Some(&candidate) = self.select_literal_single(context, candidate, variant_value) {
-      return Some(candidate)
+    if let Some(&ref candidate) = self.select_literal_single(context, candidate, variant_value) {
```

Comment:
This is called "destructuring", also known as pattern matching. I'm also destructuring the value out of the `Some`. In the case of destructuring, you can think of `&` as having the opposite effect: I'm destructuring the `&` off of the value... ie, dereferencing it. So in both of the cases here, the destructuring is about unwrapping something: the Some, and the reference.